fix(curriculum): adding missing explanations for new concepts in calorie counter project (#52312)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-form-validation-by-building-a-calorie-counter/63b61584def8fa2ebcc259e0.md

# --description--

You need to split your `str` into individual characters. You can use the <dfn>split()</dfn> method to do this.

> The `split()` method splits a string into an array of substrings, and returns the new array. You can pass in an optional separator which tells the computer where each split should happen.

Reviewer: ```suggestion
The `split()` method splits a string into an array of substrings, and returns the new array. You can pass in an optional separator which tells the method where each split should happen.
```
Being very picky here. Telling the "computer" just sounded awkward to me. But maybe I'm the only one?

> Here is an example of how to use it:

Reviewer: ```suggestion
For example, passing an empty string into the `split` method will split the string into an array of individual characters.
```

I think it is helpful to state in words what the example is demonstrating, at least to the extent possible.

```js
const str = 'Hello World';
const strArray = str.split('');
// ["H", "e", "l", "l", "o", " ", "W", "o", "r", "l", "d"]
```

> Declare a `strArray` variable and assign it that split value.

Reviewer: I think this instruction is not clear enough.

```suggestion
Split the string passed into the `cleanInputString` function into an array of individual characters and assign it to the `strArray` variable.
```

I'm guessing this could be shortened a little, feel free to do so. My main concern was that the phrase "assign it that split value" was too ambiguous.
